src/nfts/saga.ts

```diff
@@ -112,17 +116,108 @@ export function* findCelebratedNft({ payload: { nfts } }: PayloadAction<FetchNft
     celebratedNftFound({
       networkId: celebratedNft.networkId,
       contractAddress: celebratedNft.contractAddress,
+      expirationDate: expirationDateString,
+      rewardReminderDate: rewardReminderDateString,
+      deepLink,
     })
   )
 }
 
+export function* findNftReward({ payload: { nfts } }: PayloadAction<FetchNftsCompletedAction>) {
+  const featureGateEnabled = getFeatureGate(StatsigFeatureGates.SHOW_NFT_CELEBRATION)
+  if (!featureGateEnabled) {
+    return
+  }
+
+  const {
+    celebratedNft,
+    expirationDate: expirationDateString,
+    rewardReminderDate: rewardReminderDateString,
+    deepLink,
+  } = getDynamicConfigParams(DynamicConfigs[StatsigDynamicConfigs.NFT_CELEBRATION_CONFIG])
+
+  if (
+    !celebratedNft ||
+    !celebratedNft.networkId ||
+    !celebratedNft.contractAddress ||
+    !expirationDateString ||
+    !rewardReminderDateString ||
+    !deepLink
+  ) {
+    return
+  }
+
+  const expirationDate = Date.parse(expirationDateString)
+  const rewardReminderDate = Date.parse(rewardReminderDateString)
+
+  if (Number.isNaN(expirationDate)) {
+    Logger.error(TAG, 'Invalid expiration date in remote config')
+    return
+  }
+
+  if (Number.isNaN(rewardReminderDate)) {
+    Logger.error(TAG, 'Invalid reminder date in remote config')
+    return
+  }
+
+  const expired = isPast(expirationDate)
+  if (expired) {
+    return
+  }
+
+  const userOwnsCelebratedNft = nfts.some(
+    (nft) => isSameNftContract(nft, celebratedNft) && !!nft.metadata
+  )
+  if (!userOwnsCelebratedNft) {
+    return
+  }
+
+  // TODO: check if the user is in allowed list (see ACT-1100)
+
+  const lastNftCelebration = yield* select(nftCelebrationSelector)
+  if (!lastNftCelebration) {
+    return
+  }
+
+  if (!isSameNftContract(lastNftCelebration, celebratedNft)) {
```

Comment:
i found this function still quite hard to understand because there are so many early return conditions. i think there's room to simplify this, e.g. the checks for whether network id / contract address are already done in `isSameNftContract`, so we can remove some of the early return conditions relating to that? the bottom sheet itself also checks if the user owns the nft, so perhaps that condition can also be removed? it'd be ideal to have a condition that is the negative of all these early returns but idk if that's possible